packages/gatsby/src/redux/persist.ts

```diff
@@ -1,12 +1,155 @@
+import path from "path"
 import v8 from "v8"
-import { readFileSync, writeFileSync } from "fs-extra"
-import { ICachedReduxState } from "./types"
+import {
+  existsSync,
+  mkdtempSync,
+  readFileSync,
+  removeSync,
+  renameSync,
+  writeFileSync,
+} from "fs-extra"
+import { IReduxNode, ICachedReduxState } from "./types"
+import { sync as globSync } from "glob"
 
-const file = (): string => `${process.cwd()}/.cache/redux.state`
+const legacyCacheFile = path.join(process.cwd(), `.cache/redux.state`)
+const reduxCacheFolder = path.join(process.cwd(), `.cache/redux`)
+function reduxRestFile(dir: string): string {
+  return path.join(dir, `redux.rest.state`)
+}
+function reduxChunkFilePrefix(dir: string): string {
+  return path.join(dir, `redux.node.state_`)
+}
+
+function readFromLegacyCache(): ICachedReduxState {
+  // TODO: remove this bit with next major bump (fairly benign change but still)
```

Comment:
```suggestion
  // TODO: remove this bit in v3 (fairly benign change but still)
```
we don't have codified rules to be able to grep things we will want to change (neither keep track of it in some issue), so let's use `v3` as keyword for notes like that (we already use it that in other places). This might not be completely accurate (i.e. we might keep some of behaviour marked with "remove in v3", but at least we can search for them easier)

Reply:
Sure, good point